src/structures/paging/mapper/mod.rs

```diff
@@ -105,6 +191,27 @@ pub trait Mapper<S: PageSize> {
         flags: PageTableFlags,
     ) -> Result<MapperFlush<S>, FlagUpdateError>;
 
+    /// Updates the flags of an existing mapping.
+    fn update_flags_p4_entry(
+        &mut self,
+        page: Page<S>,
+        flags: PageTableFlags,
+    ) -> Result<MapperFlush<S>, FlagUpdateError>;
+
+    /// Updates the flags of an existing mapping.
+    fn update_flags_p3_entry(
+        &mut self,
+        page: Page<S>,
+        flags: PageTableFlags,
+    ) -> Result<MapperFlush<S>, FlagUpdateError>;
+
+    /// Updates the flags of an existing mapping.
+    fn update_flags_p2_entry(
+        &mut self,
+        page: Page<S>,
+        flags: PageTableFlags,
+    ) -> Result<MapperFlush<S>, FlagUpdateError>;
+
```

Comment:
I think we should name these methods "set" instead of "update" to make it more clear that they overwrite the entry flags (in contrast to ORing the flags). Also, I think I'm in favor of naming the methods `set_pX_entry_flags` instead of `set_flags_pX_entry`, but this is only personal preference.